node.go

```diff
@@ -64,16 +64,17 @@ var _ node.DecodeBlockFunc = DecodeBlock
 // Therefore, `bytes.Equal(b, Decode(b).RawData())` may not hold. If you already
 // have a CID for this data and want to ensure that it doesn't change, you
 // should use `DecodeBlock`.
+// mhType is multihash code to use for hashing, for example mh.SHA2_256
 //
 // Note: This function does not hold onto `b`. You may reuse it.
-func Decode(b []byte) (*Node, error) {
+func Decode(b []byte, mhType uint64) (*Node, error) {
```

Comment:
However, I'm wary about changing these interfaces as that will break a lot of code.